podman/domain/images_manager.py

```diff
@@ -306,22 +300,25 @@ def pull(
                 raise ValueError("'auth_config' requires keys 'username' and 'password'")
             params["credentials"] = f'{username}:{password}'
 
-        if "tls_verify" in kwargs:
-            params["tlsVerify"] = kwargs.pop("tls_verify")
+        params["tlsVerify"] = kwargs.get("tls_verify")
 
         response = self.client.post("/images/pull", params=params)
-        body = response.json()
 
         if response.status_code != 200:
+            body = response.json()
             raise APIError(response.url, response=response, explanation=body["error"])
 
-        if not all_tags:
-            return self.get(body["images"][0])
+        for item in response.iter_lines():
+            body = json.loads(item)
+            if all_tags and "images" in body:
+                images: List[Image] = []
+                for name in body["images"]:
```

Comment:
Is `body["images"]` guaranteed to always exist in the JSON and always be a list?  If it's a string or some other iteratable, the loop will still function but any error eventually given to the user may not make much sense.  Maybe it's worth type-checking and throwing an exception.  This will help direct any debugging effort closer to the actual problem area (JSON input processing) here.

Reply:
The check `"images" in body` ensures the key exists, the get() below will fail if the type isn't a list.